lib/system/ansi_c.nim

```diff
@@ -92,18 +95,40 @@ when defined(macosx):
 elif defined(haiku):
   const SIGBUS* = cint(30)
 
-when defined(nimSigSetjmp) and not defined(nimStdSetjmp):
+# "nimRawSetjmp" is defined by default for certain platforms, so we need the
+# "nimStdSetjmp" escape hatch with it.
+when defined(nimSigSetjmp):
   proc c_longjmp*(jmpb: C_JmpBuf, retval: cint) {.
     header: "<setjmp.h>", importc: "siglongjmp".}
-  template c_setjmp*(jmpb: C_JmpBuf): cint =
+  proc c_setjmp*(jmpb: C_JmpBuf): cint =
```

Comment:
The compiler croaked on that template, so I made it a proc. It might be a resurrection of this old issue: https://github.com/nim-lang/Nim/issues/5225 - but I couldn't make a small test case for a new one.

It also shows how important it is to have these defines in the test suite.